### PR TITLE
Optimizes tgui-next initial open

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

image
https://cdn.discordapp.com/attachments/326831214667235328/636044334210482177/2019-10-21_20-02-40.mp4
tgui next opt is with this optimization
Now faster than most nano implementations, and only a frame slower compared to bay's
Removes a couple calls here that were eating a lot of time in the initial render, and kills an unneeded var that wasn't even really used.
100-150% performance gains

## Why It's Good For The Game

much fast

## Changelog
:cl:
refactor: Refactored tgui-next's initial open resulting in it now opening 2-2.5x as fast
/:cl:

### CODE REVIEW

**Review by stylemistake — APPROVED** [review]

File: code/modules/tgui/tgui.dm [hunk 1]
```diff
@@ -121,12 +121,11 @@
 
 	// Remove titlebar and resize handles for a fancy window
 	// right from the beginning
-	var/list/config_data = get_config_data();
-	if(config_data["fancy"])
-		window_options["titlebar"] = FALSE
```
Comment (AnturK) [inline code/modules/tgui/tgui.dm]: What's the point of these now if they don't reflect desired client-side state.

Comment (actioninja) [reply code/modules/tgui/tgui.dm]: Not sure what you mean. This was just an unwrapping of what that previous codeblock did.

Comment (AnturK) [reply code/modules/tgui/tgui.dm]: You're not setting window_options now, which is not a local variable.

Comment (actioninja) [reply code/modules/tgui/tgui.dm]: Oh I see what you mean, I didn't realize window_options was a tgui level var. Going to test something on this real quick.

Comment (actioninja) [reply code/modules/tgui/tgui.dm]: After some further testing and benchmarking, the usage of window_options here wasn't actually what was killing performance it was the get_config_data call. Old usage can be preserved with no observable changes in performances. Not sure how needed window_options is considering it's only used here and only these two parameters are used though. Could save some memory by killing it.

Comment (actioninja) [reply code/modules/tgui/tgui.dm]: Also that get_config_data sucks performance that hard might be worth looking into at some point later as well.

**Comment by TheVekter** [comment]
If this is the one you posted the video of where it opens in like... a tenth of a second, _god yes please now_

**Comment by actioninja** [comment]
Yes, this is that change https://cdn.discordapp.com/attachments/326831214667235328/636044334210482177/2019-10-21_20-02-40.mp4

**Review by stylemistake — APPROVED** [review]

**Comment by stylemistake** [comment]
This shit is great.

**Comment by stylemistake** [comment]
Needs rebasing.

**Comment by stylemistake** [comment]
Rebased this for you.

**Comment by actioninja** [comment]
you got rid of a couple of changed I had made after whenever you had cloned.

**Pull request merged**